views/js/pciCreator/dev/audioRecordingInteraction/runtime/js/recorder.js

```diff
@@ -107,6 +118,34 @@ define([
             analyser,               // the WebAudio node use to read the input level
             frequencyArray;         // used to compute the input level from the current array of frequencies
 
+        /**
+         * Check if client uses the iOS device.
+         *
+         * @param {Array} frequencyArray
+         * @returns {*|boolean}
+         */
+        function isIOSDevice(frequencyArray) {
+            return frequencyArray.length && /(iPhone|iPad)/i.test(navigator.userAgent)
+        }
+
+        /**
+         * Fill the frequencyArray with fake data
+         *
+         * @param {Array} frequencyArray
+         */
+        function fillFakeEmitter(frequencyArray) {
+            var frequencyLength = frequencyArray.length;
+
+            frequencyArray[0] = getRndInteger(0, frequencyLength * frequencyLength);
+
+            _.forEach(frequencyArray, function(level, index) {
+                frequencyArray[index] = frequencyArray[0];
+                if (frequencyLength / 2 < index) {
+                    return false;
```

Comment:
I still fill uncomfortable with this algo. Only half of the array is filled with values, and the remaining is left as it. Maybe a comment explaining the intention would help.

And since this is loop over a fixed count, wouldn't it be better to use `_.times()` instead?
```javascript
_.times(frequencyLength / 2, function(index) {
    frequencyArray[index] = frequencyArray[0];
});
```

Reply:
I got you idea of more concise code, but decided to apply plain old vanilla in favour of speed.